pkg/cliplugins/workspace/cmd/workspace.go

```diff
@@ -0,0 +1,570 @@
+			return errors.New("No previous workspace exists !")
+		}
+		_, workspaceName = getScopeAndName(string(o.startingConfig.Contexts[kcpPreviousWorkspaceContextKey].Cluster))
+		delete(o.startingConfig.Contexts, kcpPreviousWorkspaceContextKey)
+	}
+
+	currentWorkspaceScope, currentWorkspaceName, err := o.getCurrentWorkspace()
+	if err == nil && currentWorkspaceName != "" {
+		o.startingConfig.Contexts[kcpPreviousWorkspaceContextKey] = &api.Context{
+			Cluster: currentWorkspaceScope + "/" + currentWorkspaceName,
+		}
+	}
+
+	workspaceKubeConfigBytes, err := tenancyClient.RESTClient().Get().Resource("workspaces").SubResource("kubeconfig").Name(workspaceName).Do(context.TODO()).Raw()
+	if err != nil {
+		return err
+	}
+
+	workspaceConfig, err := clientcmd.Load(workspaceKubeConfigBytes)
+	if err != nil {
+		return err
+	}
+
+	workspaceConfigCurrentContext := workspaceConfig.CurrentContext
+	scope, _ := getScopeAndName(workspaceConfigCurrentContext)
+	workspaceContextName := kcpWorkspaceContextNamePrefix + workspaceConfigCurrentContext
+
+	currentContextName := o.startingConfig.CurrentContext
+	if o.kubectlOverrides.CurrentContext != "" {
+		currentContextName = o.kubectlOverrides.CurrentContext
+	}
+
+	currentContext := o.startingConfig.Contexts[currentContextName]
+	var currentContextAuthInfo *api.AuthInfo
+	if currentContext != nil {
+		currentContextAuthInfo = o.startingConfig.AuthInfos[currentContext.AuthInfo]
+	}
+	o.startingConfig.Clusters[workspaceContextName] = workspaceConfig.Clusters[workspaceConfigCurrentContext]
+	workspaceAuthInfo := PrioritizedAuthInfo(&o.kubectlOverrides.AuthInfo, currentContextAuthInfo)
+	o.startingConfig.AuthInfos[workspaceContextName] = workspaceAuthInfo
+
+	o.startingConfig.Contexts[workspaceContextName] = &api.Context{
+		Cluster:  workspaceContextName,
+		AuthInfo: workspaceContextName,
+	}
+
+	o.startingConfig.CurrentContext = workspaceContextName
+
+	if _, err := o.Out.Write([]byte(fmt.Sprintf("Current %s workspace is \"%s\".\n", scope, workspaceName))); err != nil {
+		return err
+	}
+	return clientcmd.ModifyConfig(o.configAccess, *o.startingConfig, true)
+}
+
+func (o *WorkspaceOptions) getCurrentWorkspace() (scope string, name string, err error) {
+	currentContextName := o.startingConfig.CurrentContext
+	if o.kubectlOverrides.CurrentContext != "" {
+		currentContextName = o.kubectlOverrides.CurrentContext
+	}
+
+	if currentContextName == "" {
+		return "", "", errors.New("No current context !")
+	}
+
+	if !strings.HasPrefix(currentContextName, kcpWorkspaceContextNamePrefix) {
+		return "", "", errors.New("The current context is not a KCP workspace !")
+	}
+
+	scope, workspaceName := getScopeAndName(strings.TrimPrefix(currentContextName, kcpWorkspaceContextNamePrefix))
+	return scope, workspaceName, nil
+}
+
+func (o *WorkspaceOptions) checkCurrentWorkspace(workspaceName string, tenancyClient *tenancyclient.TenancyV1alpha1Client) (err error) {
```

Comment:
what does check mean? Need go doc and/or a better name.

Reply:
Will change the name to `checkWorkspaceExists`